### ISSUE TITLE
Validate payload data types

### ISSUE THREAD

**Issue by Steditor** [colyseus/command#5]
The current typing of the optional `validate` method of commands assumes that the payload data is already checked for correct data types. As incoming messages by clients must never be trusted without data type checking, there should be a canonical way of doing so, e.g.

- have the dispatcher call `validate` before setting the payload and have `validate` receive the payload as `unknown` data type. If the method returns true, the data can be trusted and is of the specified payload data type.
- add a separate (optional) method to commands for checking the payload data, that is called by the dispatcher before setting the payload.

Either way the interface of `Dispatcher.dispatch` should then allow to pass unknown data.

**Comment by JoaoCnh** [comment]
I know this is a few years late but a similar approach to the zod validators for messages could be nice

**Comment by endel** [comment]
Hi @JoaoCnh 👋 we did introduced message validation via zod on 0.17, tho the only and default behaviour so far is to close the client connection in case of validation failure - to prevent users trying to hack around.

https://docs.colyseus.io/room#message-input-validation

**Comment by JoaoCnh** [comment]
yeah I forgot with messages you already have payload validation so the command would just be validating things twice 🤦‍♂️
